scale-tests/batch_test.py

```diff
@@ -123,10 +128,28 @@ def submit_loop(submits_per_min: int, dispatchers: typing.List[typing.Dict], use
                  "--conf", "spark.mesos.driver.failoverTimeout={}".format(args["--spark-mesos-driver-failover-timeout"])
                 ]
 
+    if args["--spark-mesos-executor-gpus"]:
+        MEMORY_MULTIPLIER = 20
+        memory = int(args["--spark-mesos-executor-gpus"]) * MEMORY_MULTIPLIER
+        user_conf += ["--conf", "spark.driver.memory={}g".format(str(memory)),
+                      "--conf", "spark.executor.memory={}g".format(str(memory)),
+                      "--conf", "spark.mesos.gpus.max={}".format(args["--spark-mesos-max-gpus"]),
+                      "--conf", "spark.mesos.executor.gpus={}".format(args["--spark-mesos-executor-gpus"]),
+                      "--conf", "spark.mesos.executor.docker.image={}".format(args["--docker-image"]),
+                      "--conf", "spark.mesos.executor.docker.forcePullImage=false"
+                      ]
+        app_url = GPU_IMAGE_RECOGNITION_APP_URL
+    else:
+        app_url = MONTE_CARLO_APP_URL
+
     if args["--spark-mesos-driver-labels"] is not None:
         user_conf += ["--conf", "spark.mesos.driver.labels={}".format(args["--spark-mesos-driver-labels"])]
 
     if not args["--no-supervise"]:
         user_conf += ["--supervise"]
 
-    submit_loop(int(args["--submits-per-min"]), dispatchers, user_conf)
+    if args["--max-num-dispatchers"]:
```

Comment:
Let's move this up to where the dispatchers are first loaded so as to keep all dispatcher-related logic together. We could even move it to a function so as to not make the intent clearer.

Reply:
We also don't check whether `--max-num-dispatchers` makes sense for the specified list of dispatchers.